force-app/main/default/classes/DisplayPDFController.cls

```diff
@@ -258,37 +405,211 @@ global with sharing class DisplayPDFController
+        }
+    }
+     
+    /**
+     * @description    Saves content versions (text or image) to Salesforce as ContentVersion records
+     * @param          title           Title of the content version
+     * @param          base64DataList List of maps containing base64-encoded data and record types (text or image)
+     * @param          templateId      Id of the document template where the content versions should be associated
+     * @return         List<Map<String, String>> List of maps containing saved record types and their corresponding Ids
+     */
+   @AuraEnabled(cacheable=false)
+public static List<Map<String, String>> saveContentVersion(String title, List<Map<String, String>> base64DataList, String templateId) {
+    try {
+        
+        List<ContentVersion> contentVersionsToInsert = new List<ContentVersion>();
+        List<Map<String, String>> savedIds = new List<Map<String, String>>();
+        List<Id> contentVersionIdsToDelete = new List<Id>();
+        for (ContentVersion cv : [SELECT Id, ContentDocumentId  FROM ContentVersion WHERE FirstPublishLocationId = :templateId WITH USER_MODE]) {
+            contentVersionIdsToDelete.add(cv.ContentDocumentId);
+        }
+        List<Id> contentDocumentIdsToDelete = new List<Id>();
+        List<ContentDocument> cdlst = [SELECT Id FROM ContentDocument WHERE Id IN :contentVersionIdsToDelete WITH USER_MODE];
+        if (!cdlst.isEmpty()) {
+            delete cdlst;
+        }
+        for(Map<String, String> dataUrl : base64DataList) {
+            System.debug('inside for loop in savecontentversion');
+            String recordType = dataUrl.keySet().iterator().next();
+            String data = dataUrl.get(recordType);
+            
+            if(recordType.equalsIgnoreCase('Text') || recordType.equalsIgnoreCase('Image')) {
+                ContentVersion contentVersion = new ContentVersion(
+                    Title = dataUrl.get('title'),
+                    VersionData = EncodingUtil.base64Decode(data),
+                    PathOnClient = title + '.png',
+                    FirstPublishLocationId = templateId
+                );
+                
+                contentVersionsToInsert.add(contentVersion);
+            }
+        }
+         if (Schema.sObjectType.ContentVersion.isCreateable()) {
+            insert contentVersionsToInsert;
+            for(ContentVersion contentVersion : contentVersionsToInsert) {
+                Map<String, String> savedId = new Map<String, String>();
+                savedId.put(contentVersion.Title, contentVersion.Id);
+                savedIds.add(savedId);
+            }
+             System.debug('savedIds >> '+ savedIds);
+             return savedIds;
+         }
+        else{
+            LogHandler.createLog((Id) null,'DisplayPDFController.saveContentVersion()','Insufficient permissions to create ContentVersion records. ','Insufficient permissions to create ContentVersion records.', 'Insufficient permissions');
```

Comment:
Why is Insufficient access being hard-corded instead of using error.getMessage(). Incorrect syntax for the parameters that needs to be returned for the LogHandler.createdLog method.